packages/core-api/src/apis/definitions/auth.ts

```diff
@@ -223,3 +223,12 @@ export const githubAuthApiRef = createApiRef<OAuthApi & SessionStateApi>({
   id: 'core.auth.github',
   description: 'Provides authentication towards Github APIs',
 });
+
+/**
+ * Provides authentication for custom identity providers.
```

Comment:
Would like to know a bit more about the use-case here. The main purpose of the other providers is currently to allow delegation of 3rd party resources to backstage plugins. That is for a plugin to e.g. be able to read status of GCP resources.

Are you envisioning that this would be for delegating access to generic resources, or to be able to sign in with custom OAuth providers? And what is the purpose of the being standardized on the client-side?

What I'm thinking is that this mostly makes sense as a provider in the auth-backend, but in the frontend you'd end up with a custom implementation that is tailored to the organization that is using backstage. If you for example within a company have a FooService that allows you to manage Foos, you'd end up with a `fooAuthApiRef` here, which is used to access Foos, but in the backend it would be implemented using the generic OAuth provider.

Reply:
The main use case would be for custom OAuth providers and is intended as only generic on the backend. The frontend would indeed be tailored to the organization that is using it.  I wanted to include an example of what it would look like in the frontend, but that can be removed.  It may be better to name it `yourOrganizationApiRef ` to show that the frontend piece would be different for everyone.